src/fenics_constitutive/stress_strain.py

```diff
@@ -132,9 +138,198 @@ def strain_from_grad_u(grad_u: np.ndarray, constraint: Constraint) -> np.ndarray
             strain_view[:, 1] = grad_u_view[:, 4]
             strain_view[:, 2] = grad_u_view[:, 8]
             strain_view[:, 3] = 1 / 2**0.5 * (grad_u_view[:, 1] + grad_u_view[:, 3])
-            strain_view[:, 4] = 1 / 2**0.5 * (grad_u_view[:, 5] + grad_u_view[:, 7])
-            strain_view[:, 5] = 1 / 2**0.5 * (grad_u_view[:, 2] + grad_u_view[:, 6])
+            strain_view[:, 4] = 1 / 2**0.5 * (grad_u_view[:, 2] + grad_u_view[:, 6])
+            strain_view[:, 5] = 1 / 2**0.5 * (grad_u_view[:, 5] + grad_u_view[:, 7])
         case _:
             msg = "Constraint not supported."
             raise NotImplementedError(msg)
     return strain
+
+
+class UniaxialStrainFrom3D(IncrSmallStrainModel):
+    """
+    Convert a 3D model to a uniaxial strain model. This is achieved by copying the
+    relevant 1D components to a 3D array, calling the 3D model, and then copying the
+    3D components back to the 1D array. Since this converter creates new arrays for
+    the 3D components, it is (probably) not suitable for large-scale simulations due
+    to the memory consumption.
+
+    Args:
+        model: 3D model to convert to uniaxial strain.
+
+    Attributes:
+        model: 3D model to convert to uniaxial strain.
+        stress_3d: 3D stress array.
+        tangent_3d: 3D tangent array.
+        grad_del_u_3d: 3D array of gradient of displacement increment.
+    """
+
+    def __init__(self, model: IncrSmallStrainModel) -> None:
+        assert model.constraint == Constraint.FULL
+        self.model = model
+        self.stress_3d = None
+        self.tangent_3d = None
+        self.grad_del_u_3d = None
+
+    @property
+    def constraint(self) -> Constraint:
+        return Constraint.UNIAXIAL_STRAIN
+
+    def update(self) -> None:
+        self.model.update()
```

Comment:
I'm not sure what is to be expected from model.update(). As fare as I know wen didn't used it in our models yet. But could it be that in there also a transformation from original 3D to now 1D is necessary?

Reply:
At this point, it is probably not necessary. We used it for the UMATs before to update the time which was stored by the material class and currently it is used to update the total strain which is also stored in the UMAT-Wrapper. We could probably find a workaround for that too, but I am still hesitant to remove it

Reply:
Ok, then it's fine